packages/rx_bloc_cli/mason_templates/bricks/feature_onboarding/__brick__/lib/feature_onboarding/ui_components/phone_number_picker.dart

```diff
@@ -0,0 +1,191 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_rx_bloc/flutter_rx_bloc.dart';
+import 'package:provider/provider.dart';
+import 'package:widget_toolkit/widget_toolkit.dart';
+
+import '../../app_extensions.dart';
+import '../blocs/onboarding_phone_bloc.dart';
+import '../models/country_code.dart';
+import '../models/onboarding_phone_errors.dart';
+import '../services/phone_number_formatter.dart';
+import '../services/search_country_code_service.dart';
+
+class PhoneNumberPicker extends StatelessWidget {
+  const PhoneNumberPicker({
+    this.countryCodeTextStyle,
+    this.countryCodeBgColor = const Color(0xFFE1E1E1),
+    this.phoneNumberHint = 'XX XXX XXXX',
+    this.countryCodeHint = 'XXX',
+    super.key,
+  });
+
+  /// The text style of the country code text
+  final TextStyle? countryCodeTextStyle;
+
+  /// Background color of the country code segment
+  final Color countryCodeBgColor;
+
+  /// The hint text of the country code input
+  final String countryCodeHint;
+
+  /// The hint text of the phone number input
+  final String phoneNumberHint;
+
+  static const double _borderRadius = 16.0;
+
+  @override
+  Widget build(BuildContext context) => Padding(
+        padding: const EdgeInsets.symmetric(horizontal: 8),
```

Comment:
Please do not hardcode spacings, instead use the design system

Reply:
Done.